telemetry/cnstelemetry.go

```diff
@@ -23,14 +22,20 @@ const (
 
 // SendCnsTelemetry - handles cns telemetry reports
 func SendCnsTelemetry(interval int, reports chan interface{}, service *restserver.HTTPRestService, telemetryStopProcessing chan bool) {
-	retrieveMetadata := true
 
 CONNECT:
-	telemetryBuffer, err := NewTelemetryBuffer()
-	if err == nil {
-		go telemetryBuffer.Start(time.Duration(interval))
+	telemetryBuffer := NewTelemetryBuffer()
+	err := telemetryBuffer.StartServer()
+	if err == nil || telemetryBuffer.FdExists {
+		if err := telemetryBuffer.Connect(); err != nil {
+			log.Printf("[Telemetry] Failed to establish telemetry manager connection.")
+			time.Sleep(time.Minute * 1)
+			goto CONNECT
+		}
+
+		go telemetryBuffer.BufferAndPushData(time.Duration(0))
 
-		heartbeat := time.NewTicker(time.Minute * 30).C
+		heartbeat := time.NewTicker(time.Minute * 1).C
```

Comment:
do we want 1 min heartbeats?

Reply:
oops..i forget to revert this change. Thanks for pointing out